BDArmory/FX/ExplosionFX.cs

```diff
@@ -852,6 +855,17 @@ private void ExecutePartBlastEvent(PartBlastHitEvent eventToExecute)
                 if (rb != null && rb.mass > 0 && !BDArmorySettings.PAINTBALL_MODE)
                     AddForceAtPosition(rb, (Position - part.transform.position).normalized * eventToExecute.NegativeForce * BDArmorySettings.EXP_IMP_MOD * 0.25f, part.transform.position);
             }
+	    }
+            if (warheadType == WarheadTypes.Standard && ProjMass > 0 && realDistance <= blastRange)
+            {
+                float HitAngle = Vector3.Angle((eventToExecute.HitPoint + rb.velocity * TimeIndex - Position).normalized, -eventToExecute.Hit.normal);
+                float anglemultiplier = (float)Math.Cos(Math.PI * HitAngle / 180.0);
+                float thickness = ProjectileUtils.CalculateThickness(part, anglemultiplier);
+                thickness += eventToExecute.IntermediateParts.Select(p => p.Item3).Sum(); //add armor thickness of intervening parts, if any
+                if (BDArmorySettings.DEBUG_ARMOR) Debug.Log("[BDArmory.ExplosiveFX]: Part " + part.name + " hit by shrapnel; " + HitAngle + " deg hit, cumulative armor thickness: " + thickness);
+
+                ProjectileUtils.CalculateShrapnelDamage(part, eventToExecute.Hit, Caliber, Power, realDistance, SourceVesselName, ExplosionSource, ProjMass, -1, thickness); //part hit by shrapnel, but not pressure wave
+            }
```

Comment:
once blast calcs done, do shrapnel cacls, both for parts within blast AoE and parts simply within shrapnel range.
After my initial response to shrapnel implementation, having a day to think on it and look at the code if we wanted shrapnel checks this would be the place to put it, not PartIsInLineOfSight(), since what is needed is the cumulative thickness of intervening armor, (which we have here, without having to do a round of raycasts and culling out parts that have already been hit in the process of penning adjacent parts) to see if the shrapnel would pen deep enough to penetrate this current part (if other intermediate parts in the way)